### PR TITLE
Multipart optimization

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

**Comment by rossabaker** [comment]
Ugh, commits got tangled.  Hold on.

**Comment by rossabaker** [comment]
A'ight.

File: core/src/main/scala/org/http4s/multipart/MultipartDecoder.scala [hunk 1]
```diff
@@ -31,17 +32,17 @@ private[http4s] object MultipartDecoder {
...
         case Some((Left(headers), h1)) =>
           if (lastWasLeft){
             go(Part(Headers(part.headers.toList ::: headers.toList), EmptyBody), true)(h1)
           } else {
            Pull.output1(part) >> go(Part(headers, EmptyBody), true)(h1)
           }
-        case Some((Right(byte), h1)) =>
-          go(part.copy(body = part.body.append(Stream.emit[Task, Byte](byte))), false)(h1)
+        case Some((Right(bv), h1)) =>
+          go(part.copy(body = part.body.append(Stream.chunk(ByteVectorChunk(bv)))), false)(h1)
```
Comment (ChristopherDavenport) [inline core/src/main/scala/org/http4s/multipart/MultipartDecoder.scala]: Finally we use `ByteVectorChunk`

Comment (rossabaker) [reply core/src/main/scala/org/http4s/multipart/MultipartDecoder.scala]: I was just trying to get rid of it earlier this week!

It's also used in blaze.

**Pull request merged**